src/zarr/testing/store.py

```diff
@@ -158,6 +158,10 @@ async def test_get(self, store: S, key: str, data: bytes, byte_range: ByteReques
         """
         Ensure that data can be read from the store using the store.get method.
         """
+        if isinstance(store, ObjectStore) and not data:
```

Comment:
Would it be possible to put this skip / xfail on the `ObjectStore` subclass (and call `super().test_get(...)` for the cases you don't skip)?

Or we could remove this test case if we think it's not generally valid for all store implementations.

Reply:
Actually here we do still need this skip because the test cases are **not even partially satisfiable**. What https://github.com/apache/arrow-rs/pull/6751 fixed is that if there's a file with 100 bytes and you request range 90-110, you'll get the last 10 bytes. But these tests are akin to having a 100 byte-long file and requesting a range 110-120. That [isn't satisfiable](https://developer.mozilla.org/en-US/docs/Web/HTTP/Guides/Range_requests#partial_request_responses), and the HTTP spec says it should error with [416 Range Not Satisfiable](https://developer.mozilla.org/en-US/docs/Web/HTTP/Reference/Status/416).

Reply:
I ran into some issues with these tests in https://github.com/rapidsai/kvikio/pull/646 as well ("reading" pass the end of a file).

Removing these edge cases entirely is probably a bigger discussion, but IMO moving the skip to the subclass is worth keeping the base class agnostic to the store implementation.

Reply:
Works for me.